doc/smart_contract/nfr_ctrt_v2.md

```diff
@@ -0,0 +1,505 @@
+
+const nc_id = 'CF46ab6o5HTfyLwgyBhwrhkEmLxbaLkSJ8a';
+const nc = nftv2.NFTCtrtV2Whitelist(nc_id, ch);
+```
+
+### Querying
+
+#### Issuer
+
+The address that has the issuing right of the NFT contract instance.
+
+```javascript
+// nc: nftv2.NFTCtrtV2Whitelist
+
+console.log(await nc.getIssuer());
+```
+
+Example output
+
+```
+Addr { data: 'ATse3RcjEzwc5JHDPcduPYe4qA2mWhSNZaV' }
+```
+
+#### Maker
+
+The address that made this NFT contract instance.
+
+```javascript
+// nc: nftv2.NFTCtrtV2Whitelist
+
+console.log(await nc.getMaker());
+```
+
+Example output
+
+```
+Addr { data: 'ATse3RcjEzwc5JHDPcduPYe4qA2mWhSNZaV' }
+```
+
+#### Regulator
+
+The address that serves as the regulator of the NFT contract instance.
+
+```javascript
+// nc: nftv2.NFTCtrtV2Whitelist
+
+console.log(await nc.getRegulator());
+```
+
+Example output
+
+```
+Addr { data: 'ATse3RcjEzwc5JHDPcduPYe4qA2mWhSNZaV' }
+```
+
+#### Unit
+
+The unit of tokens defined in this NFT contract instance.
+
+As the unit is obviously fixed to 1 for NFTs, the support of querying unit of NFT is for the compatibility with other token-defining contracts.
+
+```javascript
+// nc: nftv2.NFTCtrtV2Whitelist
+
+console.log(await nc.getUnit());
+```
+
+Example output
+
+```
+1
+```
+
+#### Is user in the list
+
+Check if the user is in the whitelist/blacklist
+
+```javascript
+// acnt: Account
+// nc: nftv2.NFTCtrtV2Whitelist
+
+console.log(await nc.isUserInList(acnt.addr.data));
+```
+
+Example output
+
+```
+false
+```
+
+#### Is contract in the list
+
+Check if the user is in the whitelist/blacklist
+
+```javascript
+// nc: nftv2.NFTCtrtV2Whitelist
+arbitrary_ctrt_id = 'CEsGmTPZMvPkkG7g5gyqgRcXRVc2ZcVXz9J';
+
+console.log(await nc.isCtrtInList(nc_id));
+```
+
+Example output
+
+```
+false
+```
+
+### Actions
+
+#### Issue
+
+Define a new NFT and issue it. Only the issuer of the contract instance can take this action. The issued NFT will belong to the issuer.
+
+```javascript
+import * as nftv2 from './src/contract/nft_ctrt_v2.js';
+
+// acnt: Account
+// nc: nftv2.NFTCtrtV2Whitelist
+
+const resp = await nc.issue(acnt);
+console.log(resp);
+```
+
+Example output
+
+```
+{
+  "type":9,
+  "id":"Cfw93vP2JrKovjbefEh6RDiYNNxcmhBuEspLA6ghU7ws",
+  "fee":30000000,
+  "feeScale":100,
+  "timestamp":1646376022067759104,
+  "proofs":[
+    {
+        "proofType":"Curve25519",
+        "publicKey":"6gmM7UxzUyRJXidy2DpXXMvrPqEF9hR1eAqsmh33J6eL",
+        "address":"AU6BNRK34SLuc27evpzJbAswB6ntHV2hmjD",
+        "signature":"33fMLbFKgAKfD73KcYKmp44WFwDET9NsamviEUwrLhZFeU5H1GdBjqCSt8THWSbt8CFcWCZpaio4jEEcyem3g9z3"
+    }
+  ],
+  "contractId":"CFD6rH5xHXQqweUwAJ3B7J2owvPiKpewsRc",
+  "functionIndex":1,
+  "functionData":"12Wfh1",
+  "attachment":""
+}
+```
+
+#### Send
+
+Send an NFT to another user.
+
+```javascript
+import * as nftv2 from './src/contract/nft_ctrt_v2.js';
+
+// acnt0: Account
+// acnt1: Account
+// nc: nftv2.NFTCtrtV2Whitelist
+
+const resp = await nc.send(acnt0, acnt1.addr.data, 0);
```

Comment:
```suggestion
const resp = await nc.send(
acnt0, // by
acnt1.addr.data, // recipient
0 // tokIdx
);
```
if the parameter numbers is more than 2, we should add comments of the parameter's names. update all txs in this fashion.